core/pta/system.c

```diff
@@ -841,6 +844,28 @@ static TEE_Result system_get_tpm_event_log(uint32_t param_types,
 	return res;
 }
 
+static TEE_Result system_supp_plugin_invoke(uint32_t param_types,
+					    TEE_Param params[TEE_NUM_PARAMS])
+{
+	struct tee_supp_plugin_rpc_op op;
+	uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
+					  TEE_PARAM_TYPE_VALUE_INPUT,
+					  TEE_PARAM_TYPE_MEMREF_INOUT,
+					  TEE_PARAM_TYPE_NONE);
+
+	if (exp_pt != param_types)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	tee_uuid_from_octets(&op.uuid, (const uint8_t *)params[0].memref.buffer);
+
+	op.cmd = params[1].value.a;
+	op.sub_cmd = params[1].value.b;
+	op.data = params[2].memref.buffer;
+	op.len = params[2].memref.size;
+
+	return tee_invoke_supp_plugin_rpc(&op);
```

Comment:
I wouldn't mind open coding `tee_invoke_supp_plugin_rpc()` inside this function instead. It would simplify things a little.

Reply:
Do you mean to move all `tee_invoke_supp_plugin_rpc()` implementation to here?
In this case it seems we can't use plugins from optee kernel.

I assumed using `tee_invoke_supp_plugin_rpc()` directly in kernel and the new libutee API in TAs
```C
TEE_Result tee_invoke_supp_plugin(const TEE_UUID *uuid, uint32_t cmd,
				  uint32_t sub_cmd, void *buf, size_t len);
```

Reply:
Do we need this in OP-TEE Core now? Adding lots of glue some an unused "nice to have" seems wasteful.

Reply:
> Do we need this in OP-TEE Core now? 

Of course, mainly, the plugin feature will be used in TAs, but we really use it in OP-TEE Core too.
So far we have one real case using the plugins from the core:
We use a deferred-work framework to run some jobs as soos as possible the tee-supplicant starts. Some of these jobs use the plugin framework.

Reply:
OK